src/ImageData/CartaFitsImage.h

```diff
@@ -0,0 +1,105 @@
+   SPDX-License-Identifier: GPL-3.0-or-later
+*/
+
+//# CartaFitsImage.h : FITS Image class derived from casacore::ImageInterface for images not supported by casacore,
+//# including compressed and Int64
+
+#ifndef CARTA_BACKEND_IMAGEDATA_CARTAFITSIMAGE_H_
+#define CARTA_BACKEND_IMAGEDATA_CARTAFITSIMAGE_H_
+
+#include <casacore/casa/Utilities/DataType.h>
+#include <casacore/images/Images/ImageInterface.h>
+#include <casacore/lattices/Lattices/TiledShape.h>
+
+#include <fitsio.h>
+
+#include "../Logger/Logger.h"
+
+namespace carta {
+
+class CartaFitsImage : public casacore::ImageInterface<float> {
+public:
+    // Construct an image from a pre-existing file.
+    CartaFitsImage(const std::string& filename, unsigned int hdu = 0);
+    // Copy constructor
+    CartaFitsImage(const CartaFitsImage& other);
+    ~CartaFitsImage() override;
+
+    // implement casacore ImageInterface
+    casacore::String imageType() const override;
+    casacore::String name(bool stripPath = false) const override;
+    casacore::IPosition shape() const override;
+    casacore::Bool ok() const override;
+    casacore::DataType dataType() const override;
+    casacore::Bool doGetSlice(casacore::Array<float>& buffer, const casacore::Slicer& section) override;
+    void doPutSlice(const casacore::Array<float>& buffer, const casacore::IPosition& where, const casacore::IPosition& stride) override;
+    const casacore::LatticeRegion* getRegionPtr() const override;
+    casacore::ImageInterface<float>* cloneII() const override;
+    void resize(const casacore::TiledShape& newShape) override;
+    casacore::uInt advisedMaxPixels() const override;
+    casacore::IPosition doNiceCursorShape(casacore::uInt maxPixels) const override;
+
+    // implement functions in other casacore Image classes
+    casacore::Bool isMasked() const override;
+    casacore::Bool hasPixelMask() const override;
+    const casacore::Lattice<bool>& pixelMask() const override;
+    casacore::Lattice<bool>& pixelMask() override;
+    casacore::Bool doGetMaskSlice(casacore::Array<bool>& buffer, const casacore::Slicer& section) override;
+
+    casacore::DataType internalDataType() const;
+
+private:
+    // Uses _fptr (nullptr when file is closed)
+    fitsfile* OpenFile();
+    void CloseFile();
+    void CloseFileIfError(const int& status, const std::string& error);
+
+    void SetUpImage();
+    void GetFitsHeaders(int& nkeys, std::string& hdrstr);
+
+    // casacore ImageFITSConverter workaround
+    casacore::CoordinateSystem SetCoordinateSystem(
+        int nheaders, const std::string& header_str, casacore::RecordInterface& unused_headers, int& stokes_fits_value);
+    bool AddDirectionCoordinate(casacore::CoordinateSystem& coord_sys, const ::wcsprm& wcs, std::vector<int>& direction_axes);
+    bool AddStokesCoordinate(casacore::CoordinateSystem& coord_sys, const ::wcsprm& wcs, const casacore::IPosition& shape, int& stokes_axis,
```

Comment:
The `stokes_fits_value` and `stokes_axis` may be swapped in the header function definition?

Reply:
Fixed it, thanks for catching this.